public/app/features/canvas/runtime/element.tsx

```diff
@@ -122,98 +125,126 @@ export class ElementState implements LayerElement {
+        // delete placement.left;
+        break;
+      case HorizontalConstraint.LeftRight:
+        placement.left = placement.left ?? 0;
+        placement.right = placement.right ?? 0;
+        // style.left = `${placement.left}px`;
+        // style.right = `${placement.right}px`;
+        // delete placement.width;
+        style.width = '';
+        break;
+      case HorizontalConstraint.Center:
+        placement.left = placement.left ?? 0;
+        placement.width = placement.width ?? 100;
+        // translate[0] = '-50%';
+        // style.left = `calc(50% - ${placement.left}px)`;
+        style.width = `${placement.width}px`;
+        // delete placement.right;
+        break;
+      case HorizontalConstraint.Scale:
+        placement.left = placement.left ?? 0;
+        placement.right = placement.right ?? 0;
+        // style.left = `${placement.left}%`;
+        // style.right = `${placement.right}%`;
+        // delete placement.width;
+        style.width = '';
+        break;
+    }
+    this.options.placement = placement;
+
+    let transformY = '0px';
+    if (vertical === VerticalConstraint.Bottom) {
+      transformY = `${sceneHeight! - (placement.bottom ?? 0) - (placement.height ?? 100)}px`;
+    } else if (vertical === VerticalConstraint.Top) {
+      transformY = `${placement.top ?? 0}px`;
+    } else if (vertical === VerticalConstraint.TopBottom) {
+      transformY = `${placement.top ?? 0}px`;
+      style.height = `${sceneHeight! - (placement.top ?? 0) - (placement.bottom ?? 0)}px`;
+    } else if (vertical === VerticalConstraint.Center) {
+      transformY = `${sceneHeight! / 2 - (placement.top ?? 0) - (placement.height ?? 0) / 2}px`;
+    } else if (vertical === VerticalConstraint.Scale) {
+      transformY = `${(placement.top ?? 0) * (sceneHeight! / 100)}px`;
+      style.height = `${sceneHeight! - (placement.top ?? 0) * (sceneHeight! / 100) - (placement.bottom ?? 0) * (sceneHeight! / 100)}px`;
+    }
+
+    let transformX = '0px';
+    if (horizontal === HorizontalConstraint.Right) {
+      transformX = `${sceneWidth! - (placement.right ?? 0) - (placement.width ?? 100)}px`;
+    } else if (horizontal === HorizontalConstraint.Left) {
+      transformX = `${placement.left ?? 0}px`;
+    } else if (horizontal === HorizontalConstraint.LeftRight) {
+      transformX = `${placement.left ?? 0}px`;
+      style.width = `${sceneWidth! - (placement.left ?? 0) - (placement.right ?? 0)}px`;
+    } else if (horizontal === HorizontalConstraint.Center) {
+      transformX = `${sceneWidth! / 2 - (placement.left ?? 0) - (placement.width ?? 0) / 2}px`;
+    } else if (horizontal === HorizontalConstraint.Scale) {
+      transformX = `${(placement.left ?? 0) * (sceneWidth! / 100)}px`;
+      style.width = `${sceneWidth! - (placement.left ?? 0) * (sceneWidth! / 100) - (placement.right ?? 0) * (sceneWidth! / 100)}px`;
+    }
+    style.transform = `translate(${transformX}, ${transformY}) rotate(${placement.rotation ?? 0}deg)`;
```

Comment:
Let's be sure to consolidate this inside the switch statements. Awesome job getting this to work so intuitively!